### PR TITLE
Prevent ineligible vendor from getting paid.

### LINKED ISSUE #574: Prevent payment to vendor who is not a small business for auctions with start price > $3500
As an admin I want to be prevented from paying a vendor via CAP integration for auctions with starting price > $3500 if that vendor is not a small business (This case could potentially arise if the vendor's status changes during the auction) So the system should prevent that from happening

Scenario: System prevents payment to vendor who is not a small business for auctions > $3500
Given that I am logged in as an admin
And there is an auction with a start price > $3500
And the contract was awarded to a vendor
And the vendor has delivered the work
And I have accepted the work via the UI https://micropurchase-staging.18f.gov/admin/auctions/[id]/edit (select accepted from result dropdown)
But Sam.gov tells me that the vendor is no longer a small business
Then I see an error message "This vendor is not a small business and is not qualified to receive the award"
And the payment request is not sent to CAP
And the auction result is rejected

Note: We may be overloading the "Accepted" field - is accepting the deliverable/paying the vendor two separate things? Let's see what happens

### PR DESCRIPTION
**Pull request by jessieay**
* Closes #574
* Added spec for Admin::AuctionsController#update
* Removed `let` from Admin::AuctionsController specs

Opening as new version of https://github.com/18F/micropurchase/pull/630/

### CODE REVIEW

File: spec/services/update_auction_spec.rb [hunk 1]
```diff
@@ -88,47 +91,10 @@
...
 
-        updater = UpdateAuction.new(auction, params)
+        UpdateAuction.new(auction, params)
 
         expect(CreateCapProposalJob).to_not have_received(:perform_later).with(auction.id)
       end
     end
   end
-
-  describe '#should_create_cap_proposal?' do
```
Comment (jessieay) [inline spec/services/update_auction_spec.rb]: this is now a private method

Comment (adelevie) [inline spec/services/update_auction_spec.rb]: as it should be 💯

File: app/services/update_auction.rb [hunk 2]
```diff
@@ -1,20 +1,40 @@
-class UpdateAuction < Struct.new(:auction, :params, :user)
+class UpdateAuction
+  def initialize(auction, params)
+    @auction = auction
+    @params = params
+  end
+
   def perform
     auction.assign_attributes(attributes)
+    create_purchase_request
```
Comment (harrisj) [inline app/services/update_auction.rb]: Do you want to have something up here where it returns if the cap_request_url has been set?

File: app/services/update_auction.rb [hunk 3]
```diff
@@ -1,20 +1,40 @@
...
-    auction.save
   end
 
   def should_create_cap_proposal?
     auction_accepted_and_cap_proposal_is_blank? &&
       winning_bidder_is_eligible_to_be_paid?
   end
 
-  private
+  def vendor_ineligible?
```
Comment (adelevie) [inline app/services/update_auction.rb]: solid naming

**Comment by adelevie** [comment]
LGTM. I'll merge when 📗

File: app/services/update_auction.rb (hunk 2 above)
Comment (jessieay) [inline app/services/update_auction.rb]: I don't think so?

**Comment by jessieay** [comment]
@adelevie 🍏

**Pull request merged**